File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from math import sqrt
from itertools import permutations

def check(n):
    if n < 2:
        return False
    
    for i in range (2, int(sqrt(n))+1):
        if n % i == 0:
            return False
    return True

def solution(numbers):
    answer = []
    numbers = list(numbers)
    temp = []
    
    for i in range(1, len(numbers) + 1):
        temp += list(permutations(numbers, i))
    num = [int(''.join(t)) for t in temp]
    
    for i in num:
        if check(i):
            answer.append(i)
    
    return len(set(answer))

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

bool isPrime(int a) {
    if (a == 0 || a == 1)
        return false;
    
    for (int i = 2; i < a; i++) {
        if (a % i == 0) return false;
    }
    
    return true;
}

int solution(string numbers) {
    int answer = 0;
    sort(numbers.begin(), numbers.end());
    unordered_set<int> set;
    
    do {
        string str = "";
        for (int i = 0; i < numbers.size(); i++) {
            str += numbers[i];
            int num = stoi(str);
            if (isPrime(num))
                set.insert(num);
        }
        
    }while(next_permutation(numbers.begin(), numbers.end()));
    
    
    return set.size();
}